content_script.js

```diff
@@ -22,105 +20,99 @@ $(function() {
+        var start_end_date  = $(this).find('span[id*="MTG_DATES"]').text();
 
-        start_date      = getDateString(start_end_date);
-        end_date        = getDateString(start_end_date.substr(13));
+        var start_date      = getDateString(start_end_date);
+        var end_date        = getDateString(start_end_date.substr(13));
 
         // Start the event on the day before start_date, then exclude it in an exception date rule
         // This ensures an event does not occur on start_date if start_date is not on part of days_of_week
-        date_before     = start_date - 1;
+        var date_before     = start_date - 1;
 
-        start_end_times = days_times.match(/\d\d?:\d\d[AP]M/g);
+        var start_end_times = days_times.match(/\d\d?:\d\d[AP]M/g);
 
         if (start_end_times) {
-          start_time    = getTimeString(start_end_times[0]);
-          end_time      = getTimeString(start_end_times[1]);
-
-          days_of_week  = getDaysOfWeek(days_times.match(/[A-Za-z]* /)[0])
-
-          ics_content = "BEGIN:VEVENT\n" +
-                        "DTSTART:" + formatDateTime(date_before, start_time) + "\n" +
-                        "DTEND:" + formatDateTime(date_before, end_time) + "\n" +
-                        "LOCATION:" + room + "\n" +
-                        "RRULE:FREQ=WEEKLY;UNTIL=" + formatDateTime(end_date, end_time) + ";BYDAY=" + days_of_week + "\n" +
-                        "EXDATE:"   + formatDateTime(date_before, start_time) + "\n" +
-                        "SUMMARY:"  + course_code + component + ' in ' + room + "\n" +
-                        "DESCRIPTION:" +
-                          'Course Name: '    + course_name + '\\n' +
-                          'Section: '        + section + '\\n' +
-                          'Instructor: '     + instructor + '\\n' +
-                          'Component: '      + component + '\\n' +
-                          'Class Number: '   + class_number + '\\n' +
-                          'Days/Times: '     + days_times + '\\n' +
-                          'Start/End Date: ' + start_end_date + '\\n' +
-                          'Location: '       + room + '\\n\n' +
-                        "END:VEVENT\n";
+          var start_time    = getTimeString(start_end_times[0]);
+          var end_time      = getTimeString(start_end_times[1]);
+
+          var days_of_week  = getDaysOfWeek(days_times.match(/[A-Za-z]* /)[0])
+
+          var ics_content = 'BEGIN:VEVENT\n' +
+                            'DTSTART;TZID=' + time_zone + ':' + formatDateTime(date_before, start_time) + '\n' +
+                            'DTEND;TZID=' + time_zone + ':' + formatDateTime(date_before, end_time) + '\n' +
+                            'LOCATION:' + room + '\n' +
+                            'RRULE:FREQ=WEEKLY;UNTIL=' + formatDateTime((parseInt(end_date) + 1).toString(), end_time) + 'Z;BYDAY=' + days_of_week + '\n' +
```

Comment:
Kind of hacky, since `end_date + 1` isn't necessarily valid (e.g. `20140131 + 1`), but something similar is already done on L89, so it's good enough.